### PR TITLE
Send Findgroup if we lose a close group node

### PR DESCRIPTION
**Pull request by mmoadeli**
addresses issue #252

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/431)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
Is this work in progress or ready for review?

File: src/routing_membrane.rs [hunk 1]
```diff
@@ -388,10 +388,10 @@ impl<F> RoutingMembrane<F> where F: Interface {
         // The relay map will automatically drop the Name if the last endpoint to it is dropped
         self.relay_map.remove_unknown_connection(&endpoint);
         self.relay_map.drop_endpoint(&endpoint);
-        let mut trigger_handle_churn = false;
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: why do you rename this?

File: src/routing_membrane.rs [hunk 2]
```diff
@@ -434,6 +435,34 @@ impl<F> RoutingMembrane<F> where F: Interface {
...
+        let serialised_msg = match encode(&find_group_msg) {
+            Ok(msg) => msg,
+            Err(_) => panic!("Failure to serialise find group message.")
+        };
+
+        if self.routing_table.size() != 0 {
+            return self.send_swarm_or_parallel(&our_name, &serialised_msg);
+        }
+
+        loop {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: This is not correct within address relocation; so please limit the function send_find_group to only do what the function says. nothing more.

also, if you are making a send_find_group function, update all cases where we send find_group_request consistently

File: src/routing_membrane.rs [hunk 3]
```diff
@@ -434,6 +435,34 @@ impl<F> RoutingMembrane<F> where F: Interface {
...
+    /// If routing table is not empty, the find group request is sent through available node(s),
+    /// else if bootstrap_endpoint is non-empty, request sent through bootstrap_endpoint.
+    /// If routing table is empty & bootstrap_endpoint is empty, attempt is made to
+    /// send the request through the endpoint acheived from re-bootstrapping.
+    fn send_find_group_request(&mut self) {
+        let our_name = self.own_name.clone();
+        let find_group_msg = self.construct_find_group_msg(&our_name);
+        let serialised_msg = match encode(&find_group_msg) {
+            Ok(msg) => msg,
+            Err(_) => panic!("Failure to serialise find group message.")
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: just as below, have the function return a result that we can ignore( ) for now, but throw an error, don't panic

File: src/routing_membrane.rs [hunk 4]
```diff
@@ -434,6 +435,34 @@ impl<F> RoutingMembrane<F> where F: Interface {
...
+        };
+
+        if self.routing_table.size() != 0 {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: you don't need to check this; the function is simply a no-op; and no bootstrapping connections should be used by default

File: src/routing_membrane.rs [hunk 5]
```diff
@@ -434,6 +435,34 @@ impl<F> RoutingMembrane<F> where F: Interface {
         };
     }
 
+    /// If routing table is not empty, the find group request is sent through available node(s),
+    /// else if bootstrap_endpoint is non-empty, request sent through bootstrap_endpoint.
+    /// If routing table is empty & bootstrap_endpoint is empty, attempt is made to
+    /// send the request through the endpoint acheived from re-bootstrapping.
+    fn send_find_group_request(&mut self) {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: additionally, send_ functions are collected below, please group them where they are

File: src/routing_membrane.rs [hunk 6]
```diff
@@ -434,6 +435,34 @@ impl<F> RoutingMembrane<F> where F: Interface {
...
+        let our_name = self.own_name.clone();
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: you don't need to clone this

**Pull request closed without merging**